pkg/karmadactl/interpret/interpret.go

```diff
@@ -0,0 +1,166 @@
+
+        - retention                : (desired|desired-file), (observed|observed-file)
+        - replicaResource          : (desired|desired-file|observed|observed-file)
+        - replicaRevision          : (desired|desired-file|observed|observed-file), desired-replica
+        - statusReflection         : (desired|desired-file|observed|observed-file)
+        - statusAggregation        : (desired|desired-file|observed|observed-file), status-file...
+        - healthInterpretation     : (desired|desired-file|observed|observed-file)
+        - dependencyInterpretation : (desired|desired-file|observed|observed-file)
+`)
+
+	interpretExample = templates.Examples(`
+        # Check the customizations in file
+        %[1]s interpret -f customization.json --check
+        # Check the retention rule
+        %[1]s interpret -f customization.yml --operation retention --check
+    
+        # Execute the retention rule for local resource
+        %[1]s interpret -f customization.yml --operation retention --desired-file desired.json --observed-file observed.json
+        # Execute the retention rule for server resource
+        %[1]s interpret -f customization.yml --operation retention --desired-file desired.json --observed deployments/foo`)
+)
+
+const (
+	customizationResourceName = "resourceinterpretercustomizations"
+)
+
+// NewCmdInterpret new interpret command.
+func NewCmdInterpret(f util.Factory, parentCommand string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &Options{
+		IOStreams: streams,
+		Rules:     allRules,
+	}
+	cmd := &cobra.Command{
+		Use:                   "interpret (-f FILENAME | NAME) (--operation OPERATION) [-n NAMESPACE] [--ARGS VALUE]... ",
+		Short:                 "Validate and test interpreter customization before applying it to the control plane",
+		Long:                  interpretLong,
+		SilenceUsage:          true,
+		DisableFlagsInUseLine: true,
+		Example:               fmt.Sprintf(interpretExample, parentCommand),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Annotations: map[string]string{
+			util.TagCommandGroup: util.GroupClusterTroubleshootingAndDebugging,
+		},
+	}
+
+	flags := cmd.Flags()
+	options.AddKubeConfigFlags(flags)
+	flags.StringVarP(options.DefaultConfigFlags.Namespace, "namespace", "n", *options.DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
+	flags.StringVar(&o.Operation, "operation", o.Operation, "The interpret operation to use. One of: ("+strings.Join(o.Rules.Names(), ",")+")")
+	flags.BoolVar(&o.Check, "check", false, "Validates the given ResourceInterpreterCustomization configuration(s)")
+	flags.StringVar(&o.Desired, "desired", o.Desired, "The type and name of resource on server to use as desiredObj argument in rule script. For example: deployment/foo.")
```

Comment:
We don't need the flag now, right? since we are not going to introduce the functionality to check the configuration on the server, isn't it?

Reply:
We don't check the configuration on the server, but can use the resources on the server as arguments for interpreter. E.g.

```
interpret -f customization.json --operation replicaResource --observed deployments/foo --namespace=bar
```

How do you think of it.

Reply:
I don't think so. It looks great, but the `observed` should come from member clusters, not `karmada-apiserver`.

Reply:
Get it.